src/unix/core.c

```diff
@@ -983,3 +987,43 @@ int uv__dup2_cloexec(int oldfd, int newfd) {
     return r;
   }
 }
+
+
+int uv_homedir(char* home, size_t* len) {
+
+  char login[LOGIN_NAME_MAX];
+  struct passwd pw;
+  struct passwd* result;
+  char* buf;
+  size_t loginsize = sizeof(login);
+  size_t bufsize = sysconf(_SC_GETPW_R_SIZE_MAX);
+  int r;
+
+  if (home == NULL || *len == 0)
+    return -EINVAL;
+
+  r = getlogin_r(login, loginsize);
+
+  if (r != 0)
+    return -r;
+
+  buf = malloc(bufsize);
+
+  if (buf == NULL)
+    return -ENOMEM;
+
+  r = getpwnam_r((const char*) login, &pw, buf, bufsize, &result);
+  free(buf);
+
+  if (r != 0)
+    return -r;
+
+  if (result == NULL)
+    return -ENOENT;
+
+  strncpy(home, (const char*) pw.pw_dir, *len);
```

Comment:
Use after free? The buffer that stores this information is released a couple of lines up.